### ISSUE TITLE
ppss.kr tracking

### ISSUE THREAD

**Issue by JobcenterTycoon** [uBlockOrigin/uAssets#9717]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`https://ppss.kr/`

### Describe the issue

browser fingerprinting (canvas, audio, domRect, browser info like screen, cpu, ...)

### uBlock Origin version

1.37.2

### Browser name and version

Firefox developer

### Settings

- Enabled uBlock filters – Annoyances

### Notes

with
`*$script,redirect-rule=noopjs,domain=ppss.kr`
`@@||ppss.kr^$generichide`
i get way less requests

**Comment by uBlock-user** [comment]
Not seeing any slew of network requests there...explain what you see as per the logger.

**Comment by JobcenterTycoon** [comment]
xhr `https://ppss.kr/hyBqrFr6m5jeT5.otf` (its random and the end is random too). Logger output:
<img width="332" alt="Unbenannt" src="https://user-images.githubusercontent.com/48647394/128612325-c3cefc11-a404-44b9-9bae-44f91be231e0.PNG">


Data sended to the site: `MkkQR0dbVhAIEAdXAwtUU1FTH1ABC1QfBgtWUB8KAQUCHwIDBAQHBFdWAQUKVxAeEF1AW1VbXBAIEEJCQUEcWUAQHhBHQF4QCBBaRkZCQQgdHUJCQUEcWUAdEB4QW0FtU1ZQXl1RWRAIVFNeQVceEFtBbVBeXVFZW1xVbVNRUVdCRlNQXldtU1ZBEAhUU15BVx4QVFtcVVdAQkBbXEYQCBACUFQHBFQAVAsLAlcGAwsLUwpTBgFUAwsHUAECU1EAVxBP`

when i remove the script (not the base64 encoded script from adshield, i mean the tracking script on top of the site) with a filter like `ppss.kr##^script:has-text(getPrototypeOf)` the request with the data inside no longer appears (Data showed in the console not in the logger). Just 3 requests from Adshield are left but with no data.

<img width="370" alt="Unbenannt" src="https://user-images.githubusercontent.com/48647394/128612397-023fd201-7e4c-430f-9e03-323a7d9ff749.PNG">

When you see way more requests add the filter from the issue to get rid of them. Its just the encoded ad data.

**Comment by uBlock-user** [comment]
Those XHRs are used to for detecting adblock, blocking them will cause adblock detection which will trigger the anti-adblock script to throw Error contructor which will cause in aborting of site load. Adding  `ppss.kr##^script:has-text(getPrototypeOf)` doesn't affect anything I still see 8+ XHRs being fired as per the logger/

> browser fingerprinting (canvas, audio, domRect, browser info like screen, cpu, ...)

You have to yet to show that happening in any XHR that's fired there.

**Comment by JobcenterTycoon** [comment]
Ok here a better option to see that:
When i defuse ad shield with `ppss.kr##^script:has-text(createElementNS)` all ad shield XHRs are gone. Just one is left with hashed data sended the the server.

<img width="545" alt="Unbenannt" src="https://user-images.githubusercontent.com/48647394/128630004-32731bb4-9185-4602-ade1-6453567100e2.PNG">

**Comment by uBlock-user** [comment]
> ppss.kr##^script:has-text(createElementNS)

Breaks thumbnails on the homepage.

**Comment by uBlock-user** [comment]
I have to yet to see a filter you post that defuses ad-shield XHRs without triggering anti-adblock or breaking the site. So closing as wontfix.

**Comment by JobcenterTycoon** [comment]
The first one was only to show the tracking. Real filter:

defuse tracking xhr
`ppss.kr##+js(aeld, DOMContentLoaded)`

defuse ad xhr without triggering anti adblock
`ppss.kr##[href*="https://www.classo.me"]:remove()`
`ppss.kr##.ppss-title-banner:remove()`

i get no breakage with this

**Comment by uBlock-user** [comment]
Tracking is completely disabled if I add both, do you see any site breakages with both of being added ?

**Comment by JobcenterTycoon** [comment]
I tested: search, articles, images and the twitter embed and i have no breakage

**Comment by uBlock-user** [comment]
https://github.com/uBlockOrigin/uAssets/commit/315764f15438aff1a6c11fac6ff33ac4e7b66e69

**Comment by JobcenterTycoon** [comment]
Looks like they not spezific fixed the old filter because the other 2 still working

Whats about: `ppss.kr##+js(aopr, XMLHttpRequest)` ?

**Comment by uBlock-user** [comment]
@JobcenterTycoon same issue at `ygosu.com` ?

**Comment by JobcenterTycoon** [comment]
Looks like. But the script works a bit different because the filter `ygosu.com##+js(aopr, XMLHttpRequest)` is enough to block the ads and all xhrs

**Comment by uBlock-user** [comment]
I'm able to block ads but images also get blocked due to adblock detection..

**Comment by JobcenterTycoon** [comment]
Yes me too. I can’t find a working filter at the moment

**Comment by uBlock-user** [comment]
That doesn't make images appear, anyways I went with `:style` rule.

**Comment by JobcenterTycoon** [comment]
`ygosu.com` need new filters. possible new filters:

`ygosu.com##.left_ban,.right_ban,div[style="padding: 1px; display: flex; flex-direction: row; position: absolute; right: 0; top: 0; opacity: 0.75;"]:style(transform: scale(0.000001,0.000001))`
`ygosu.com##div:not(.header, .gnb, #contain, #footer, .board_body):has(:scope >span > span > span):style(transform: scale(0.000001,0.000001))`

**Comment by shadow-b1ock** [comment]
The current `ygosu.com` filter using `transform` does not work (it is working for `ppss.kr`). It triggers an adblock detection, as a result, they do not load images. I think you would want to remove the previous filters because it breaks website's necessary functionality.

I created an extension to block ads for `ppss.kr` and `ygosu.com`. Try: https://github.com/shadow-b1ock/shadowBlock

Although it does the opposite: it does not touch network resources, it merely hides the ads.

**Comment by JobcenterTycoon** [comment]
blocking the ads is not the problem the problem is they watch and patch the fix. I mean i still know working filter for `ygosu.com` but it only works for 1-2 days

**Comment by uBlock-user** [comment]
@shadow-b1ock Sure I will remove the filters for `ygosu.com` if you're going to be working on that site everytime they upgrade their detection.

**Comment by shadow-b1ock** [comment]
> blocking the ads is not the problem the problem is they watch and patch the fix. I mean i still know working filter for ygosu.com but it only works for 1-2 days. Hope they not targeting your extension

That is exactly why I created an extension. It uses more robust logic to avoid adblock detection, so it will require more hard work from ad-shield. In fact it is supposed to be used with other adblocker, it works best when it's used with ublock, minus the `transform` filters that I've reported hence I'm here.

**Comment by shadow-b1ock** [comment]
> We're doing the same thing.

I am refering to the original post, where the author's main interest seemed to be about preventing websites from firing tracking requests.

> The breakage is caused by the anti-adblock script which executes -- throw new Error('Failed to load website due to adblock.'); That's self sabotage, they're the ones breaking their own site, not us.

And _that_'s because your filters are being detected. They are such a nefarious ad agent that tries to mislead users to believe that their adblockers is breaking the site. I have written about their practice in detail in README.

**Comment by uBlock-user** [comment]
@shadow-b1ock Have you deofuscated their anti-adblock script which is inline ?

**Comment by shadow-b1ock** [comment]
No, that seemd to be another pesky byproduct of [javascript-obfuscator](https://github.com/javascript-obfuscator/javascript-obfuscator), the existence of that project must be removed from the Earth.

A common approach would be to wrap native API calls and logging its arguments. The logic is publicly available as a source code of the extension but I prefer not to disclose further, because of spies watching us.

**Comment by uBlock-user** [comment]
I'm aware of them using shadowDOM for creating ad-elements that cannot be blocked by a conventional blocker, but uBO is capable of hiding and manipulating elements present inside a shadowDOM.

**Comment by shadow-b1ock** [comment]
That's true. However, uBO cannot watch for DOM change inside shadow DOM. Its procedural cosmetic filtering engine is based on watching DOM change via MutationObservers, but change inside shadow DOM won't be notified and `querySelector` won't return nodes inside shadow DOM.

To watch shadow document trees, it requires a bomb which is rather complex to be implemented as a scriptlet.. you can see how it's done in the source code.

**Comment by uBlock-user** [comment]
@shadow-b1ock Do you mind creating a Firefox extension too ? I don't use Chromium.

**Comment by shadow-b1ock** [comment]
Well, to quickly implement the extension I used [Complex :not()](https://www.chromestatus.com/feature/5014164156186624). 
By doing `img:not(:root *)`, it selects nodes only in a shadow DOM, which is a neat trick I think. If such a selector works in Firefox, I think there would be no problem in porting the extension to FF.

**Comment by uBlock-user** [comment]
@shadow-b1ock https://bugzilla.mozilla.org/show_bug.cgi?id=933562

**Comment by shadow-b1ock** [comment]
It needs to be tested that such a selector works when it is used in an _injected stylesheets_.
Currently there are still rough edges, and clearing these will be next few updates.

**Comment by avotoko** [comment]
Hide ads and block tracking. It works at the moment. Ask them to check error for xhr.
```
||ygosu.com^$xhr,1p
@@||ygosu.com/*/$xhr,1p
```

**Comment by JobcenterTycoon** [comment]
`ppss.kr$xhr` works too.

**Comment by JobcenterTycoon** [comment]
ygosu is broken again

**Comment by shadow-b1ock** [comment]
1. Now blocking obfuscated XHR requests triggers their adblock detection, which the current rule `||ygosu.com^$xhr,1p` does, so the current 3 xhr rules should be removed.

2. >My old redirect filter still work

  I confirm this,
```
*$script,redirect-rule=noopjs,domain=ygosu.com
*$frame,redirect-rule=noopframe,domain=ygosu.com
```
makes their script to behave as if no adblock is detected. They still fire a few tracking obfuscated XHRs in this case.

3. Redirecting obfuscated XHRs to any non-empty resources does not trigger adblock detection, and breaks their script. 
```
/.+(?<!\.(yg|xml))$/$xhr,1p,domain=ygosu.com,redirect=noopjs
```
4. ShadowBlock still works. 
   - It is [live](https://chrome.google.com/webstore/detail/shadowblock/jdchbieocpofgppblcgibeckljdgocfh) in Chrome web store now.

**Comment by IMayKnow** [comment]
I encountered ads

**Comment by JobcenterTycoon** [comment]
When you mean the "sponsored" tab: `ppss.kr##section[id^="custom_html-"]:has-text(SPONSORED)`

**Comment by IMayKnow** [comment]
I may have seen yesterday adshield advertising (at least 1)

**Comment by shadow-b1ock** [comment]
`/.+(?<!\.(yg|xml))$/$xhr,1p,domain=ygosu.com,redirect=noopjs` rule is being detected. I do not need `@@||ygosu.com$ghide` on my end.

> I may have seen yesterday adshield advertising (at least 1)

Current uBO rules work by acting as if adblock is not enabled, so if you are using any other adblocker or some other filters, it may trigger their replacement ads. 

They also tried to detect [ShadowBlock](https://github.com/shadow-b1ock/shadowBlock) by another vulnerability. I have updated it to 2.5 and it works just fine.

**Comment by JobcenterTycoon** [comment]
`@@||ygosu.com$ghide` is needed in Chrome on my end they testing for `.adsbygoogle`

**Comment by shadow-b1ock** [comment]
Another attempt to detect [ShadowBlock](https://github.com/shadow-b1ock/shadowBlock) is deployed. I am guessing that they are doing a timing attack. I have updated it to 2.6 and it works just fine.

> Do you mind creating a Firefox extension too ? I don't use Chromium.
It is now available in Firefox. It uses different code because it fully utilizes Gecko-specific Xray vision APIs.

**Comment by JobcenterTycoon** [comment]
Ads are back on both sites. Wow ads no breakage. Working filters

`ppss.kr###genesis-content > span:style(position: absolute; transform: rotateX(90deg)!important)`
`ppss.kr###genesis-content > div:not([class="home-top widget-area"]):style(position: absolute; transform: rotateX(90deg)!important)`
`ppss.kr##.wing_banner:style(position: absolute; transform: rotateX(90deg)!important)`
`ppss.kr###custom_html-3:style(position: absolute; transform: rotateX(90deg)!important)`

`@@||tab2.clickmon.co.kr/$script,domain=ygosu.com`
`@@||static.dable.io/dist/plugin.min.js$script,domain=ygosu.com`

**Comment by avotoko** [comment]
> ```
> @@||tab2.clickmon.co.kr/$script,domain=ygosu.com
> @@||static.dable.io/dist/plugin.min.js$script,domain=ygosu.com
> ```
> 

On my end, the filters for ygosu.com works on some pages and not on others .
Not work: `https://ygosu.com/community/real_article/yeobgi/1855762/?page=0&frombest=Y`

The filter for ppss.kr works. But in my case, I don't subscribe to YousList, so I need `ppss.kr###custom_html-2`.

**Comment by avotoko** [comment]
The following filters works for me.
```
!#if env_firefox
ygosu.com,ppss.kr##+js(aost, XMLHttpRequest.prototype.send, /th:\d+\s+[a-zA-Z]\d\/<@[^\t]+\t\s[a-zA-Z]\d@/)
!#endif
!#if env_chromium
ygosu.com,ppss.kr##+js(aost, XMLHttpRequest.prototype.send, /th:\d+\s+in[a-zA-Z\d:;]+\t\s[a-zA-Z]\d\si/)
!#endif
! KOR: YousList includes the following filter.
ppss.kr###custom_html-2
```

**Comment by uBlock-user** [comment]
Why different filters for chrome and firefox ?

**Comment by avotoko** [comment]
> Why different filters for chrome and firefox ?

I created the firefox filter first, then the chrome filter. I just left that thought path. There is nothing wrong with having one.

**Comment by avotoko** [comment]
They have changed the code.
I've tried to follow the direction JobcenterTycoon suggested, and it seems to work.
```
! Remove
! Not work
ppss.kr,ygosu.com#@#+js(aost, XMLHttpRequest.prototype.send, /th:\d+\s+[a-zA-Z]\d\/<@[^\t]+\t\s[a-zA-Z]\d@|th:\d+\s+in[a-zA-Z\d:;]+\t\s[a-zA-Z]\d\si/)
! Cause adblock detection 
ppss.kr#@#.ppss-title-banner:remove()
ppss.kr#@#[href*="https://www.classo.me"]:remove()
ppss.kr#@#.ppss-title-banner:style(height: 1px; width: 1px; transform:translate(-999999px,0))
ppss.kr#@#[href="https://www.classo.me"]

! Add
! Allow the scripts to create the basic dom structure for ad.
@@||pagead2.googlesyndication.com/pagead/js/adsbygoogle.js$script,domain=ygosu.com|ppss.kr
@@||pagead2.googlesyndication.com/pagead/managed/js/adsense/*/show_ads_impl_*.js$script,domain=ygosu.com|ppss.kr
@@||ad.doubleclick.net/ddm/adj/Bkne/Bdfs$script,domain=ygosu.com|ppss.kr
@@||static.dable.io/dist/plugin.min.js$script,domain=ygosu.com|ppss.kr
@@||adservice.google.com/adsid/integrator.js$script,domain=ygosu.com|ppss.kr
@@||tab2.clickmon.co.kr/pop/wp_ad_*_js.php$script,domain=ygosu.com|ppss.kr
@@||adservice.google.co.jp/adsid/integrator.js$script,domain=ygosu.com|ppss.kr
@@||www.googletagservices.com/activeview/js/current/osd.js$script,domain=ygosu.com|ppss.kr
@@||api.dable.io/plugin/services/ppss.kr/prefs$script,domain=ygosu.com|ppss.kr
! And more
*$frame,redirect-rule=noopframe,domain=ppss.kr
@@||ppss.kr^$ghide
! Fix YousList
@@/^[^:]+:\/\/+([^:\/]+\.)?\x79\x67\x6f\x73\x75\x2e\x63\x6f\x6d\/([\x61-\x7a_\/]*[\x41-\x5a]+)+[\x61-\x7a_\/]*\.[0-9\x61-\x7a]+$/$image,match-case,domain=ygosu.com|ppss.kr
! title banner: On my end, ubo log shows redirected to 1x1.gif, but still displayed.
||ppss.kr/wp-content/uploads/*/ktx_750x10*$image,domain=ppss.kr
||ppss.kr/wp-content/uploads/*/classo_750x10*$image,domain=ppss.kr
! Privacy?
||connect.facebook.net^$script,domain=ppss.kr
```
Sometimes the adblock is detected or not even on the same page, I think this is a timing issue of whether the external ad script is loaded or not when the ad element is checked by anti-adblock.
The problem of remaining ad spaces and not working on m.ygosu.com is an issue for the future.

**Comment by shadow-b1ock** [comment]
- They are patching `RegExp.prototype.test` to neutralize `abort-on-stack-trace` scriptlet. One way to proceed is to fix scriptlets to keep a reference of native `RegExp.prototype.test`.
    - ShadowBlock is built from the very beginning to be resistant to such attacks.
  
 - AdGuard Extra is currently working.
    - In past, the websites were checking whether the `prototype` object is defined on certain APIs, which forced an update of ShadowBlock. It seems to be removed for now, but simply restoring that for certain APIs will detect AdGuard Extra.
   
 - ShadowBlock from version 2.8 is more resistant to timing attacks. Lower versions were working on Firefox, but were detected on Chrome; This is presumably because Firefox by default has reduced timer precision to protect against exactly this kind of attacks. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now#reduced_time_precision

**Comment by shadow-b1ock** [comment]
Just changing `a-zA-Z` in the current rules to `A-Za-z` will bypass their check.

```
ygosu.com,ppss.kr##+js(aost, XMLHttpRequest.prototype.send, /th:\d+\s+[A-Za-z]\d\/<@[^\t]+\t\s[A-Za-z]\d@|th:\d+\s+in[A-Za-z\d:;]+\t\s[A-Za-z]\d\si/)
```

**Comment by uBlock-user** [comment]
Them patching `.test()` also affects any scriptlet which makes use of it - https://github.com/uBlockOrigin/uAssets/issues/10082

**Comment by JobcenterTycoon** [comment]
For a short term solution

`ppss.kr##div[style="position: relative; display: inline-block; text-align: center;"]:style(position: absolute; transform: rotateX(90deg)!important)`

`ygosu.com##div[style="position: relative; display: inline-block; text-align: center;"]:style(position: absolute; transform: rotateX(90deg)!important)`

To counter YousList breakage: `@@/^[^:]+:\/\/+([^:\/]+\.)?\x70\x70\x73\x73\x2e\x6b\x72\/([\x61-\x7a_\/]*[\x41-\x5a]+)+[\x61-\x7a_\/]*\.[0-9\x61-\x7a]+$/$image,match-case,domain=ygosu.com|ppss.kr`

**Comment by avotoko** [comment]
>For a short-term solution, you can change a-zA-Z in the current rules to A-Za-z.

Yes, we can block ads by doing so. And, they have changed the way xhr handles errors, so the content images are removed. We can also prevent the removal of images using aost, but it is still a short term solution.

**Comment by shadow-b1ock** [comment]
I see, I didn't notice that they are detecting errors thrown from `XMLHttpRequest.prototype.send`.

It will be a trivial code change to make `aost` work with `XMLHttpRequest.prototype.status`. The more fundamental issue is that these stack-trace-based-reflection is susceptible to messing with V8-only [`Error.prepareStackTrace`](https://v8.dev/docs/stack-trace-api#customizing-stack-traces) API, although I doubt they will ever implement such a complicated measure.

**Comment by avotoko** [comment]
>It will be a trivial code change to make `aost` work with `XMLHttpRequest.prototype.status`.

Yes, I have an experimental scriptlet that traps access to such properties.

>I doubt they will ever implement such a complicated measure.

I think so too. If it is implemented, it can still be handled by uBlock/assets/resources/scriptlets.js.

**Comment by shadow-b1ock** [comment]
> If it is implemented, it can still be handled by uBlock/assets/resources/scriptlets.js.

`Error.prepareStackTrace` allows scripts to encrypt stack traces while it is being executed, so it fundamentally undermining extension's ability of inspecting stack traces within API hooks. The property `prepareStackTrace` is evaluated in runtime. Evil script can define this property to be non-configurable. If you define it before evil script, they can detect it.

My understanding is that `uBlock/assets/resources/scriptlets.js` is limited to relatively simple measures with widespread usage, and is also limited to what developers and contributors of uBO can take the burden of maintaining it ([#10082](https://github.com/uBlockOrigin/uAssets/issues/10082#issuecomment-927877891)).

Speaking of possibility, even the whole functionality of ShadowBlock can be integrated into `scriptlets.js` with a minor change so that uBO supply a secret token to scriptlets. But `scriptlets.js` hasn't even yet guarded Proxies against `Function.prototype.toString`.

**Comment by uBlock-user** [comment]
> My understanding is that uBlock/assets/resources/scriptlets.js is limited to relatively simple measures with widespread usage, and is also limited to what developers and contributors of uBO can take the burden of maintaining it (#10082).

Scriptlets are added based multiple use-cases or how helpful they are based on the scope, as for exploits, scriptlets are not exploit-proof, they're simple snippets of javascripts which help in dealing with anti-user behaviour found in the wild.

**Comment by shadow-b1ock** [comment]
Protecting scriptlets from exploits (more precisely, __prototype polution__) is certainly doable, in fact, ShadowBlock is a demonstration of it.

**Comment by avotoko** [comment]
I think knowing the flow of their system will help us to take countermeasures, so I'll summarize what I think.

1. Delete the `src` attribute of the content images from the original html on the proxy server. Save the deleted `src` attributes separately.
2. In the browser, first perform a shallow adblock check.
3. If it fails the check, the script will exit without loading the content images.
4. If the check is passed, the content images will be displayed using the saved src attribute.
5. Next they will do a deep adblock check.
6. If the check fails, they will try to load their alternative ads.
7. If the alternative ads fails to load they will remove the content images.


We can back this up with changes in the browser display.
First, the content images loads even if the site's ads are hidden. This is evidence that they are only doing a shallow check.
If the ads are hidden, the alternative ads will be displayed after a considerable delay. This is because deep checks are used to detect ad hiding.

I think the reason why the adblock check is set to two levels is to reduce the delay in displaying content images.

**Comment by shadow-b1ock** [comment]
- I doubt that this case can be dealt with uBO even if the prototype pollution vulnerability https://github.com/uBlockOrigin/uBlock-issues/issues/1741 is fixed.
    - As per the labeling scheme, this issue should be re-opened and marked as "can't fix".

 - AdGuard Extra is working.
   - They monkey-patch `Element.prototype.attachShadow`, observe shadow DOM tree, and delete elements from the DOM.   
   - It is unclear to me why they have stopped countering AG Extra, which would only need them a trivial amount of work.

 - [ShadowBlock](https://github.com/shadow-b1ock/shadowBlock) is working.

**Comment by uBlock-user** [comment]
> As per the labeling scheme, this issue should be re-opened and marked as "can't fix".

We don't label issues as "can't fix" if we have already fixed the issue multiple times, this is a battle of attrition, hence no point in keeping it open, whoever's interested in a fix can spend time on it and keep it updated.

**Comment by shadow-b1ock** [comment]
For whoever interested in can discover the issue, it should remain opened.

I fail to see why this issue is different in any sense from the currently open https://github.com/uBlockOrigin/uAssets/issues/8246. 
The issue on `aternos.org` has been fixed multiple times (https://github.com/uBlockOrigin/uAssets/issues/1494, https://github.com/uBlockOrigin/uAssets/pull/7208), and it is still unresolved.

**Comment by uBlock-user** [comment]
> For whoever interested in can discover the issue, it should remain opened.

We keep issues open as for the team to work on it or for gorhill to work on it, not for others.

About aternos, it can't be fixed with uBO as there's one specific value setting that cannot be done with uBO as there's no scriptlet or uBO functionality that can handle that. Cases like aternos require userScripts. might as well close it too.

**Comment by shadow-b1ock** [comment]
> We keep issues open as for the team to work on it or for gorhill to work on it, not for others.

I see.

> it can't be fixed with uBO

By the way, `aternos.org` case can be countered with uBO, so that's simply not true. I once investigated it out of curiosity of various anti-adblock strategies found in the wild.

**Comment by uBlock-user** [comment]
> aternos.org case can be countered with uBO, so that's simply not true.

Like I said -- "there's no scriptlet or uBO functionality that can handle that."

**Comment by uBlock-user** [comment]
> By the way, aternos.org case can be countered with uBO, so that's simply not true.

If you have filters, suggest them here and I will add it.

**Comment by shadow-b1ock** [comment]
Sorry, I don't have it right now. What I can tell is that they are listening to every xhr errors, so you will need a growing list of `$redirect=noopjs` filters and url exception filters. They are not (yet) doing something similar to the subject of this issue, `ppss.kr`, so that will be enough. 

Details need to be worked out by your team, but your team will surely be able to find a set of working filters.

**Comment by shadow-b1ock** [comment]
Oh, one more thing, `aternos.org` will display a very helpful error code in their anti-adb screen, so once you know how that code is produced, it is easy to debug what part of the filters need to be fixed.

**Comment by uBlock-user** [comment]
Added filters from AdGuard, lets see how long it lasts.

**Comment by shadow-b1ock** [comment]
Last time I checked it, filters from AdGuard were out-of-date. Also let me clarify that that issue is beyond my interest, making sure the filters are working is your team's issue.

**Comment by JobcenterTycoon** [comment]
@shadow-b1ock your addon not working anymore pictures are hidden on both sites. For anyone want a fix without breakage:

```
@@||ppss.kr^$document,csp
ppss.kr#@##ppss_wing_banner_left
ppss.kr#@#.custom-html-widget.textwidget

*$image,redirect-rule=32x32.png,domain=ygosu.com
ygosu.com,ppss.kr##div[style="position: relative; display: inline-block; text-align: center;"]:style(left: 10000px !important)
```

**Comment by JobcenterTycoon** [comment]
@uBlock-user   @Yuki2718 

tracking coming from `https://cdn.jsdelivr.net/npm/datbpzjvws@latest/dist/script.min.js` and i see xhr requests to the domain `https://www.1ee085aa-4c05-43a8-a4fe-fafedbedd062.xyz/erwrTyo8bLQeYZ4` on `ygosu.com`

**Comment by Yuki2718** [comment]
@JobcenterTycoon I can't reproduce.

**Comment by JobcenterTycoon** [comment]
Seems to be this issue is not fixable because the tracking xhr can’t be blocked without breakage or detection

**Comment by JobcenterTycoon** [comment]
@Yuki2718 can you add the fix https://github.com/uBlockOrigin/uAssets/commit/856a407317c38743f4d00c9f3cfeb727558ab651 to AG too please?

**Comment by Yuki2718** [comment]
> can you add the fix

I don't see these requests on either `ppss.kr` or `ygosu.com`.

**Comment by JobcenterTycoon** [comment]
@Yuki2718 sorry i forgot to say its on `https://tgd.kr`

**Comment by gaeulbyul** [comment]
I found that [there are more Ad-Shield scripts at npm](https://www.npmjs.com/~sanghyeon.jeon), How about blocking all of them preemptively?

```
||jsdelivr.net/npm/datbpzjvws@latest/dist/script.min.js
||jsdelivr.net/npm/sgrzjqnhfk@latest/dist/script.min.js
||jsdelivr.net/npm/mlwgycqisx@latest/dist/script.min.js
||jsdelivr.net/npm/ufxwsgjgev@latest/dist/script.min.js
||jsdelivr.net/npm/fxpmzkxstt@latest/dist/script.min.js
||jsdelivr.net/npm/rffchggias@latest/dist/script.min.js
||jsdelivr.net/npm/mrnndputqc@latest/dist/script.min.js
||jsdelivr.net/npm/lllbmbclsi@latest/dist/script.min.js
||jsdelivr.net/npm/wlfqlefvqf@latest/dist/script.min.js
||jsdelivr.net/npm/ncghpuuwdw@latest/dist/script.min.js
```

**Comment by piquark6046** [comment]
https://github.com/uBlockOrigin/uAssets/issues/9717#issuecomment-1094195224
I think that maintainers of an anti-tracking filter have to add Ad-Shield analytics source on other websites

<details>
<summary>Screenshot on loawa.com</summary>

![Screenshot from 2022-04-10 08-28-01](https://user-images.githubusercontent.com/98787049/162609723-2c1a78b9-2779-4102-9316-cece2a08ccb6.png)

</details>

https://www.ad-shield.team/solution/analytics.

**Comment by Yuki2718** [comment]
@piquark6046 Hmm, I don't see the request on `loawa.com`. I scoped domain in AGTPF as AG maintainer generally don't like such a generic regex added to uBP. Do you know other sites using the script?

**Comment by piquark6046** [comment]
https://github.com/uBlockOrigin/uAssets/issues/9717#issuecomment-1094229859
@Yuki2718 

Not yet.
However, I opened a PR.
 - https://github.com/AdguardTeam/AdguardFilters/pull/115163

**Comment by gaeulbyul** [comment]
Update: They uploaded more scripts. This time, none of packages' name are matches a `/[a-z]{10}/` pattern:

```
||cdn.jsdelivr.net/npm/ruibwv-wor-ownqwwl^
||cdn.jsdelivr.net/npm/whiou^
||cdn.jsdelivr.net/npm/vloincrw^
||cdn.jsdelivr.net/npm/noqe-jlnva^
||cdn.jsdelivr.net/npm/iknwripb-kbhewrnk^
||cdn.jsdelivr.net/npm/ubhor-phi^
||cdn.jsdelivr.net/npm/jnippij^
||cdn.jsdelivr.net/npm/bouwp^
||cdn.jsdelivr.net/npm/biyp^
||cdn.jsdelivr.net/npm/zcxqeo-bihcq-ibhw-qbue^
```

Also, There's a questionable domain hidden in the code there. (base64-encoded **`03c844c9-d265-4006-a39d-400e6cb40bb7.xyz`**)
I suspect this domain is for doing the same purpose as the domain we previously blocked before. (`1ee085aa-4c05-43a8-a4fe-fafedbedd062.xyz`)

**Comment by Yuki2718** [comment]
Where they're used? On `tgd.kr` the script is still caught by the current regex.

**Comment by piquark6046** [comment]
If you want to block Ad-Shield, please install [microShield](https://github.com/List-KR/microShield).

_This issue is mentioned at https://www.ad-shield.io/aritlcle/facebook-war._

**Comment by Daxy7** [comment]
is adshield still unblockable on this domain for example: https://github.com/easylist/easylist/issues/23071 ?

**Comment by mapx-** [comment]
https://github.com/uBlockOrigin/uAssets/commit/539cd99d43cca016a5d74f6960ac0c0bde66b18e
